Packs/Palo_Alto_Networks_WildFire/Integrations/Palo_Alto_Networks_WildFire_v2/Palo_Alto_Networks_WildFire_v2.py

```diff
@@ -402,11 +412,115 @@ def wildfire_upload_url(upload):
+      command.
+    Args:
+        outputs: the context data from the search command
+        uploaded_item: 'FILE' or 'URL'
+        args: the args initially inserted to the upload function that initiated the polling sequence
+
+    Returns:
+
+    """
+    results_function_args = {}
+    if uploaded_item == 'FILE':
+        identifier = {'md5': outputs.get('MD5')}
+    else:
+        identifier = {'url': outputs.get('URL')}
+    results_function_args.update(identifier)
+
+    results_function_args.update({key: value for key, value in args.items() if key in ['verbose', 'format']})
+    return results_function_args
+
+
+def run_polling_command(args: dict, cmd: str, upload_function: Callable, results_function: Callable, uploaded_item):
+    """
+    This function is generically handling the polling flow. In the polling flow, there is always an initial call that
+    starts the uploading to the API (referred here as the 'upload' function) and another call that retrieves the status
+    of that upload (referred here as the 'results' function).
+    The run_polling_command function runs the 'upload' function and returns a ScheduledCommand object that schedules
+    the next 'results' function, until the polling is complete.
+    Args:
+        args: the arguments required to the command being called, under cmd
+        cmd: the command to schedule by after the current command
+        upload_function: the function that initiates the uploading to the API
+        results_function: the function that retrieves the status of the previously initiated upload process
+        uploaded_item: the type of item being uploaded
+
+    Returns:
+
+    """
+    ScheduledCommand.raise_error_if_not_supported()
+    command_results_list = []
+    interval_in_secs = int(args.get('interval_in_seconds', 60))
+    # distinguish between the initial run, which is the upload run, and the results run -
+    # the 'upload' argument will only be present in the upload command args
+    is_new_search = 'upload' in args
+    if is_new_search:
+        for upload in argToList(args['upload']):
+            # narrow the args to the current single url or file
+            args['upload'] = upload
+            # create new search
+            command_results = upload_function(args)[0]
+            outputs = command_results.outputs
+            results_function_args = get_results_function_args(outputs, uploaded_item, args)
+            # schedule next poll
+            polling_args = {
+                'interval_in_seconds': interval_in_secs,
+                'polling': True,
+                **results_function_args,
+            }
+            scheduled_command = ScheduledCommand(
+                command=cmd,
+                next_run_in_seconds=interval_in_secs,
+                args=polling_args,
+                timeout_in_seconds=6000)
+            command_results.scheduled_command = scheduled_command
+            command_results_list.append(command_results)
```

Comment:
What is the maximum number of simultaneous of `ScheduledCommand`s you had running?
Try to manually test a heavy load if possible.